solver/src/solver/http_solver.rs

```diff
@@ -344,18 +347,33 @@ impl HttpSolver {
         Ok((model, context))
     }
 
-    async fn send(&self, model: &BatchAuctionModel) -> Result<SettledBatchAuctionModel> {
+    async fn send(
+        &self,
+        model: &BatchAuctionModel,
+        deadline: Instant,
+    ) -> Result<SettledBatchAuctionModel> {
+        // The timeout we give to the solver is one second less than the deadline to make up for
+        // overhead from the network.
+        let timeout_buffer = Duration::from_secs(1);
+        let timeout = deadline
+            .checked_duration_since(Instant::now() + timeout_buffer)
+            .ok_or_else(|| anyhow!("no time left to send request"))?;
+
         let mut url = self.base.clone();
         url.set_path("/solve");
 
         let instance_name = self.generate_instance_name();
         tracing::info!("http solver instance name is {}", instance_name);
         url.query_pairs_mut()
-            .append_pair("instance_name", &instance_name);
+            .append_pair("instance_name", &instance_name)
+            .append_pair("time_limit", &timeout.as_secs().to_string());
 
         self.config.add_to_query(&mut url);
         let query = url.query().map(ToString::to_string).unwrap_or_default();
-        let mut request = self.client.post(url).timeout(self.timeout);
+        // The default Client created in main has a short http request timeout which we might
+        // exceed. We remove it here because the solver already internally enforces the timeout and
+        // the driver code that calls us also enforces the timeout.
+        let mut request = self.client.post(url).timeout(Duration::from_secs(u64::MAX));
```

Comment:
Just curious: why not simply remove the ".timeout(...)" part?

Reply:
If we don't specify any timeout here it will be use the globally configured default timeout on the Client. We need to set it something to overwrite the default.